src/main/java/at/fhhagenberg/esd/sqe/ws20/gui/ECCController.java

```diff
@@ -24,248 +20,314 @@
+            rCon.setPercentHeight(100.0 / info.getNrOfFloors());
+
+            HBox hb = new HBox();
+            hb.setAlignment(Pos.CENTER_RIGHT);
+            hb.setSpacing(5);
+            hb.setPadding(new Insets(5, 5, 5, 5));
+            hb.translateXProperty().set(10);
+
+            Line line = new Line(0, 0, 10, 0);
+            line.disableProperty().bind(floors[i].isServiced.not());
+
+            Label label = new Label(Integer.toString(i));
+            label.disableProperty().bind(floors[i].isServiced.not());
+
+            hb.getChildren().addAll(
+                    createFloorImageView("images/arrowUp.png", floors[i].requestUp),
+                    createFloorImageView("images/arrowDown.png", floors[i].requestDown),
+                    createFloorImageView("images/hand.png", floors[i].stopRequest),
+                    createFloorImageView("images/filler.jpeg", targetFloor.isEqualTo(i)),
+                    label,
+                    line);
+
+            gElevatorFloors.getRowConstraints().add(rCon);
+            gElevatorFloors.add(hb, 0, info.getNrOfFloors() - i - 1);
+        }
+
+        elevators.clear();
+        for (int i = 0; i < info.getNrOfElevators(); i++) {
+            elevators.add("Elevator " + i);
+        }
+
+        timer.schedule(new TimerTask() {
+            @Override
+            public void run() {
+                update();
+            }
+        }, 0, 100);
+    }
+
+    private void update() {
+        if (model == null) {
+            log(new RuntimeException(Messages.getString("modelInvalid")));
+            return;
+        } else if (currentElevator.get() < 0)
+            return;
+
+        Platform.runLater(() -> {
+            ElevatorState elevatorState;
+
+            try {
+                elevatorState = model.queryElevatorState(currentElevator.intValue());
+            } catch (Exception e) {
+                log(e);
+                return;
+            }
+
+            speed.setValue(elevatorState.getCurrentSpeed());
+            position.setValue(elevatorState.getCurrentPosition());
+            weight.setValue(elevatorState.getCurrentWeight());
+            currentFloor.setValue(elevatorState.getCurrentFloor());
+            isDoorOpen.setValue(elevatorState.getCurrentDoorStatus() == DoorStatus.OPEN);
+            isDirectionUp.setValue(elevatorState.getCurrentDirection() == Direction.UP);
+
+            var servicedFloors = elevatorState.getServicedFloors();
+
+            for (int i = 0; i < info.getNrOfFloors(); i++) {
+                try {
+                    var state = model.queryFloorState(i);
+                    floors[i].requestUp.set(state.isUpRequest());
+                    floors[i].requestDown.set(state.isDownRequest());
+                    floors[i].stopRequest.set(elevatorState.getCurrentFloorButtonsPressed().get(i));
+                    floors[i].isServiced.set(servicedFloors.get(i));
+                } catch (Exception e) {
```

Comment:
Can the `.set()` methods above can fail? If they can, we should not directly log the exception messages for those exceptions. In the end, we should also catch our specific exceptions and not the generic one.

Reply:
resolved in 43eec4f1ddae3243847d43f115687bfe6ce00de0